### DISCUSSION TITLE
Remove backslashes '\' in the result of  json.dump()

### DISCUSSION THREAD

**Discussion by zliucd** [nlohmann/json#3374] [Q&A] [answered]
Hi,

```j``` is a ```json``` object having a list(vector) of strings. The output of ```output = j.dump(4)``` contains lots of backslashes '\'.  I wanted to save the result in file, so the file has many backslashes too. 

But when I printed the result in ``` cout<<output<<endl```,  it printed nicely. 

Is there a way to remove "\" automatically ? Thanks.

**Comment by falbrechtskirchinger** [comment]
Where are you seeing those `\"`? Are you maybe looking at the string content in a debugger?
Some characters cannot appear in a string literal verbatim and need to be escaped using a backslash. E.g. `str = "\"";` is a string containing a single quote mark.

As you say, writing to `std::cout` works fine, likewise writing to `std::ofstream` should not produce a file containing unnecessary backslash characters.

**Reply by zliucd** [reply]
```\``` appeared debugger console, and they are still visible when writing JSON strings to file.   When dumping json with indent 4 ```json.dump(4)```, there is a newline character ```\n``` before each key in the string.

I would expect ```\``` and ```\n``` are not visible in output file.

**Reply by falbrechtskirchinger** [reply]
Can you show an example (output and maybe code to reproduce)? Where exactly in the JSON output do these escape sequences appear? If they're inside string values, than that would of course be expected.

**Comment by zliucd** [comment, marked as answer]
It's solved by using ```json::array()``` to append a list of json objects and then dump.

**Reply by isaiahbank** [reply]
the answer is not clear. Can you write the full code to give clarity.
